### PR TITLE
docs(MultiSelect): add story variant of inline notification to docs

### PR DESCRIPTION
**Pull request by mcwinter07**
## Why
We released the component prior to the design docs being ready  / announced

## What
- adds an update notification
- adds a story variant of inline notifications to allow for JSX to be passed in as an arg (avoiding the MDX parser styles for p tags)

### CODE REVIEW

File: storybook/components/StoryInlineNotification/StoryInlineNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import {
+  InlineNotification,
+  InlineNotificationProps,
+} from "~components/Notification"
+
+export type StoryInlineNotificationProps = {
+  content: ReactNode
+} & InlineNotificationProps
+
+export const StoryInlineNotification = ({
```
Comment (HeartSquared) [inline storybook/components/StoryInlineNotification/StoryInlineNotification.tsx:11]: Same comments as the other PR, but I reckon just make this one a contextual component. Something like `SbNotificationUpcomingRelease`.

Comment (gyfchong) [reply storybook/components/StoryInlineNotification/StoryInlineNotification.tsx:11]: Why make it specific?

Comment (mcwinter07) [reply storybook/components/StoryInlineNotification/StoryInlineNotification.tsx]: Re-linking posted in other [PR](https://github.com/cultureamp/kaizen-design-system/pull/4395#discussion_r1414893197)

Comment (mcwinter07) [reply storybook/components/StoryInlineNotification/StoryInlineNotification.tsx]: Also what do you think of the name `SbReleaseNotification`. I think it still captures the immediacy / intent when it should be used

Comment (HeartSquared) [reply storybook/components/StoryInlineNotification/StoryInlineNotification.tsx]: Yeah open to any naming :)

Comment (mcwinter07) [reply storybook/components/StoryInlineNotification/StoryInlineNotification.tsx]: Groovy that's all updated a ready to roll :)

File: packages/components/src/MultiSelect/_docs/MultiSelect.mdx [hunk 2]
```diff
@@ -12,7 +12,10 @@ import * as MultiSelectStories from "./MultiSelect.stories"
   className="!mb-8"
 />
 
-<KaioNotification />
+<SbReleaseNotification
+  type="negative"
+  content={<span>This component has not been officially released. Please wait for the an announcement in <code>#update_design_system</code> and for the design guidance to be published to Figma and Confluence before consuming.</span>}
```
Comment (HeartSquared) [inline packages/components/src/MultiSelect/_docs/MultiSelect.mdx:17]: Ohhh. I meant bake these two into the component itself, not just rename the generic one 😂

**Review by HeartSquared — APPROVED** [review]

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**